projects/common-platform-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/client/OsClient.kt

```diff
@@ -0,0 +1,132 @@
+package uk.gov.justice.digital.hmpps.integrations.client
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.service.annotation.GetExchange
+
+interface OsClient {
+    @GetExchange("/search/places/v1/find")
+    fun searchByFreeText(
+        @RequestParam query: String,
+        @RequestParam maxResults: Int,
+        @RequestParam minMatch: Double,
+    ): OsPlacesResponse
+}
+
+data class OsPlacesResponse(
+    val header: OsPlacesHeader,
+    val results: List<DpaWrapper>?
+)
+
+data class OsPlacesHeader(
+    val uri: String?,
+    val query: String?,
+    val offset: Int?,
+    val totalresults: Int?,
+    val format: String?,
+    val dataset: String?,
+    val lr: String?,
+    val maxresults: Int?,
+    val matchprecision: Double?,
+    val filter: String?,
+    val srs: String?,
+    val epoch: String?,
+    val lastupdate: String?,
+    val output_srs: String?
```

Comment:
Minor thing - Sonar is complaining about this line: https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&branch=PI-2667&id=ministryofjustice_hmpps-probation-integration-services&open=AZObsMQVKrrBTNgO6p_o